routes/tasks.js

```diff
@@ -5,14 +5,20 @@ const knex = require('../connection');
 const queryWays = require('../services/query-ways');
 const toGeoJSON = require('../services/osm-data-to-geojson');
 
-const properties = ['id', 'way_id', 'neighbors'];
+const properties = ['id', 'way_id', 'neighbors', 'provinces'];
 
 async function getNextTask (req, res) {
   const skip = req.query.skip ? req.query.skip.split(',') : [];
+  const province = req.query.province;
   const task = await knex.select(properties)
   .from('tasks')
   .where('pending', false)
   .whereNotIn('id', skip)
+  .modify(function(queryBuilder) {
```

Comment:
Using modify with an inner conditional feels silly but I'll let it pass. Since you're reusing the function twice, better to call it something "filterByProvince" and then give it to modify: `modify(filterByProvince)`